classes/module/Module.php

```diff
@@ -3432,33 +3438,54 @@ public function isAdminLegacyContext()
      */
     public function isSymfonyContext()
     {
-        return !$this->isAdminLegacyContext() && defined('_PS_ADMIN_DIR_');
+        // Check that a container is available but not generated by PrestaShop
+        $container = $this->getContainer();
+
+        return null !== $container && !($container instanceof LegacyContainerInterface);
     }
 
     /**
-     * Access the Symfony Container if we are in Symfony Context.
-     * Note: in this case, we must get a container from SymfonyContainer class.
-     * Note: if not in Symfony context, fallback to legacy Container for FO/BO.
+     * Access a service from the container (found from the controller or the context
+     * depending on cases). It uses ContainerFinder to find the appropriate container.
      *
      * @param string $serviceName
      *
-     * @return object|false if a container is not available, it returns false
+     * @return object|false If a container is not available it returns false
+     *
+     * @throws ServiceCircularReferenceException When a circular reference is detected
+     * @throws ServiceNotFoundException When the service is not defined
+     * @throws \Exception
      */
     public function get($serviceName)
     {
-        if ($this->isSymfonyContext()) {
-            if (null === $this->container) {
-                $this->container = SymfonyContainer::getInstance();
-            }
-
-            return $this->container->get($serviceName);
+        $container = $this->getContainer();
+        if (null !== $container) {
+            return $container->get($serviceName);
         }
 
-        if ($this->context->controller instanceof Controller) {
-            return $this->context->controller->get($serviceName);
+        return false;
+    }
+
+    /**
+     * Return the container depending on the context:
+     *  - Symfony environment: container generated by the kernel
+     *  - Legacy environment admin: container generated by PrestaShop core for admin controllers
+     *  - Legacy environment front: container generated by PrestaShop core for front controllers
+     *
+     * @return ContainerBuilder|ContainerInterface|null
+     */
+    public function getContainer()
+    {
+        if (null === $this->container) {
+            $finder = new ContainerFinder($this->context);
+            try {
+                $this->container = $finder->getContainer();
+            } catch (ContainerNotFoundException $e) {
+                return null;
+            }
```

Comment:
"A method should do what it's meant to do or throw an exception". Avoid mixed return types, because it's error prone and doesn't add value.

Consider this code:

```php
$this->getContainer()->get('some.service'); // Error: call to member function on null
```

It's best to remove the try and let consumers handle exceptions:
```suggestion
     * @return ContainerInterface
     *
     * @throws ContainerNotFoundException
     */
    public function getContainer(): ContainerInterface
    {
        if (null === $this->container) {
            $finder = new ContainerFinder($this->context);
            $this->container = $finder->getContainer();
```

Reply:
Ok fine by me, I actually introduced a similar check in the `get` method but had similar trouble so it's logical to remove it here as well